File: src/main/java/com/team573/gongguri/domain/member/entity/Review.java
package com.team573.gongguri.domain.member.entity;

import com.team573.gongguri.domain.groupPurchase.entity.GroupPurchase;
import com.team573.gongguri.global.entity.BaseEntity;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Entity
@Getter
@NoArgsConstructor
@Table(name = "review")
public class Review extends BaseEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long reviewId;

    @Column(nullable = false)
    private boolean liked;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "group_id", nullable = false)
    private GroupPurchase groupPurchase;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "member_id", nullable = false)
    private Member member;

}

File: src/main/java/com/team573/gongguri/domain/review/entity/Review.kt
package com.team573.gongguri.domain.review.entity

import com.team573.gongguri.domain.grouppurchase.entity.GroupPurchase
import com.team573.gongguri.domain.member.entity.Member
import com.team573.gongguri.global.entity.BaseEntity
import jakarta.persistence.*

@Entity
@Table(name = "review")
class Review (
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	val reviewId: Long? = null,

	@JoinColumn(name = "group_id", nullable = false)
	@ManyToOne(fetch = FetchType.LAZY)
	var groupPurchase: GroupPurchase,

	@JoinColumn(name = "member_id", nullable = false)
	@ManyToOne(fetch = FetchType.LAZY)
	var member: Member,

	@Column(nullable = false)
	var liked: Boolean
) : BaseEntity() {
	override fun equals(other: Any?): Boolean {
		if (this === other) return true
		if (javaClass != other?.javaClass) return false

		other as Review

		if (reviewId != other.reviewId) return false
		if (liked != other.liked) return false
		if (groupPurchase != other.groupPurchase) return false
		if (member != other.member) return false

		return true
	}

	override fun hashCode(): Int {
		var result = reviewId?.hashCode() ?: 0
		result = 31 * result + liked.hashCode()
		result = 31 * result + groupPurchase.hashCode()
		result = 31 * result + member.hashCode()
		return result
	}
}
